src/pip/_internal/cli/base_command.py

```diff
@@ -176,6 +176,9 @@ def _main(self, args: list[str]) -> int:
         if options.debug_mode:
             self.verbosity = 2
 
+        if hasattr(options, "progress_bar") and options.progress_bar == "auto":
```

Comment:
Does a hasattr check really need to be done here or is this to guard against some type hinter / IDE warning? I don't have a good grasp on the options object, so this is a genuine question.

Reply:
Not every command accepts `--progress-bar` so the attribute is not guaranteed to exist unfortunately.